main/lsp/LSPTypechecker.cc

```diff
@@ -456,21 +456,24 @@ LSPQueryResult LSPTypechecker::query(const core::lsp::Query &q, const std::vecto
     return LSPQueryResult{move(out.second)};
 }
 
-TypecheckRun LSPTypechecker::retypecheck(LSPFileUpdates updates) const {
-    if (!updates.canTakeFastPath) {
-        Exception::raise("Tried to typecheck slow path updates with retypecheck. Retypecheck can only typecheck the "
-                         "previously typechecked version of a file.");
-    }
-
-    for (const auto &file : updates.updatedFiles) {
-        auto path = file->path();
-        auto source = file->source();
-        auto fref = gs->findFileByPath(path);
-        if (!fref.exists() || fref.data(*gs).source() != source) {
-            Exception::raise("Retypecheck can only typecheck the previously typechecked version of a file.");
-        }
+LSPFileUpdates LSPTypechecker::getNoopUpdate(std::vector<core::FileRef> frefs) const {
+    LSPFileUpdates noop;
+    noop.canTakeFastPath = true;
+    // Epoch isn't important for this update.
+    noop.versionStart = 0;
+    noop.versionEnd = 0;
+    for (auto fref : frefs) {
+        auto &index = getIndexed(fref);
+        noop.updatedFileIndexes.push_back({index.tree->deepCopy(), index.file});
+        noop.updatedFiles.push_back(gs->getFiles()[fref.id()]);
+        noop.updatedFileHashes.push_back(globalStateHashes[fref.id()]);
     }
+    return noop;
+}
 
+TypecheckRun LSPTypechecker::retypecheck(vector<core::FileRef> frefs) const {
+    LSPFileUpdates updates = getNoopUpdate(move(frefs));
+    auto workers = WorkerPool::create(0, *config->logger);
```

Comment:
I think this is the only line I'm slightly confused about, but I also don't pretend to understand the `WorkerPool` abstraction. Does this do something side-effecting? Because otherwise it looks like it's unused.

Reply:
_embarrassed look_

I cargo-culted this from my preemptive slow path changes. It's unused and has no side effects. I'll remove it prior to merging. Thanks for catching!

Reply:
Removed. I also switched to using `indexed` directly instead of getIndexed since I don't want indexedFinalGS trees.